diff --git a/cmake/DetermineTargetArchitecture.cmake b/cmake/DetermineTargetArchitecture.cmake
index 688797f..a033a51 100644
--- a/cmake/DetermineTargetArchitecture.cmake
+++ b/cmake/DetermineTargetArchitecture.cmake
@@ -23,6 +23,8 @@ function(determine_target_architecture FLAG)
       set(ARCH "x86_64")
     elseif("${MSVC_C_ARCHITECTURE_ID}" STREQUAL "ARM")
       set(ARCH "arm")
+    elseif("${MSVC_C_ARCHITECTURE_ID}" STREQUAL "ARM64")
+      set(ARCH "arm64")
     else()
       message(FATAL_ERROR "Failed to determine the MSVC target architecture: ${MSVC_C_ARCHITECTURE_ID}")
     endif()
